pkg/registry/core/pod/strategy.go

```diff
@@ -758,3 +760,113 @@ func applySchedulingGatedCondition(pod *api.Pod) {
 		Message: "Scheduling is blocked due to non-empty scheduling gates",
 	})
 }
+
+// applyAppArmorVersionSkew implements the version skew behavior described in:
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/24-apparmor#version-skew-strategy
+func applyAppArmorVersionSkew(pod *api.Pod) {
+	if pod.Spec.OS != nil && pod.Spec.OS.Name == api.Windows {
+		return
+	}
+
+	var podProfile *api.AppArmorProfile
+	if pod.Spec.SecurityContext != nil {
+		podProfile = pod.Spec.SecurityContext.AppArmorProfile
+	}
+
+	// Handle the containers of the pod
+	podutil.VisitContainers(&pod.Spec, podutil.AllFeatureEnabledContainers(),
+		func(ctr *api.Container, _ podutil.ContainerType) bool {
+			// get possible annotation and field
+			key := api.DeprecatedAppArmorAnnotationKeyPrefix + ctr.Name
+			annotation, hasAnnotation := pod.Annotations[key]
+
+			var containerProfile *api.AppArmorProfile
+			if ctr.SecurityContext != nil {
+				containerProfile = ctr.SecurityContext.AppArmorProfile
+			}
+
+			// sync field and annotation
+			if !hasAnnotation {
+				newAnnotation := ""
+				if containerProfile != nil {
+					newAnnotation = appArmorAnnotationForField(containerProfile)
+				} else if podProfile != nil {
+					newAnnotation = appArmorAnnotationForField(podProfile)
+				}
+
+				if newAnnotation != "" {
+					if pod.Annotations == nil {
+						pod.Annotations = map[string]string{}
+					}
+					pod.Annotations[key] = newAnnotation
+				}
+			} else if containerProfile == nil {
+				newField := apparmorFieldForAnnotation(annotation)
+				if errs := corevalidation.ValidateAppArmorProfileField(newField, &field.Path{}); len(errs) > 0 {
+					// Skip copying invalid value.
+					newField = nil
```

Comment:
are we skipping so the pod creation can succeed, or so the validation errors will be flagged during normal validation?

Reply:
So pod validation can succeed, since this would be tightening validation otherwise. Although, maybe since it's an immutable field and the pod would never run anyway it's OK to just fail validation?